src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -1480,6 +1482,27 @@ impl Coordinator {
         Ok(send_immediate_rows(vec![row]))
     }
 
+    pub(super) async fn sequence_inspect_shard(
+        &self,
+        session: &Session,
+        plan: InspectShardPlan,
+    ) -> Result<ExecuteResponse, AdapterError> {
+        if !session.user().is_internal() {
```

Comment:
Not thrilled about this rbac special case here, but probably sufficient for now. I can improve later.